hermes/src/store/storage/local_storage.rs

```diff
@@ -155,12 +157,507 @@ impl Storage for LocalStorage {
     }
 
     async fn store_accumulator_state(&self, state: super::AccumulatorState) -> Result<()> {
-        let key = state.slot;
-        self.accumulator_cache.insert(key, state);
+        let mut accumulator_cache = self.accumulator_cache.write().await;
+        accumulator_cache.push_back(state);
+
+        let mut i = accumulator_cache.len().saturating_sub(1);
+        while i > 0 && accumulator_cache[i - 1].slot > accumulator_cache[i].slot {
+            accumulator_cache.swap(i - 1, i);
+            i -= 1;
+        }
+
+        if accumulator_cache.len() > self.cache_size as usize {
+            accumulator_cache.pop_front();
+        }
+
         Ok(())
     }
 
     async fn fetch_accumulator_state(&self, slot: Slot) -> Result<Option<super::AccumulatorState>> {
-        Ok(self.accumulator_cache.get(&slot))
+        let accumulator_cache = self.accumulator_cache.read().await;
+        match accumulator_cache.binary_search_by_key(&slot, |state| state.slot) {
+            Ok(idx) => Ok(accumulator_cache.get(idx).cloned()),
+            Err(_) => Ok(None),
+        }
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use {
+        super::*,
+        crate::store::{
+            proof::wormhole_merkle::WormholeMerkleMessageProof,
+            types::{
+                AccumulatorMessages,
+                ProofSet,
+            },
+        },
+        pyth_sdk::UnixTimestamp,
+        pythnet_sdk::{
+            accumulators::merkle::MerklePath,
+            hashers::keccak256_160::Keccak160,
+            messages::{
+                Message,
+                PriceFeedMessage,
+            },
+        },
+    };
+
+    pub fn create_dummy_price_feed_message_state(
```

Comment:
(if you made this take an array of `times_and_slots: [(i64, Slot)]` you could shorten the tests below a bit)

Reply:
I tried this now but the test readability was not very good. I decided to keep it as it is.